### PR TITLE
Feature/bottom sheet modal

### PR DESCRIPTION
**Pull request by andriibodryi**

### CODE REVIEW

File: src/components/BottomSheetModal/BottomSheetModal.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  const renderContent = () => (
+    <View style={styles.content}>
+      {children || <Text>Swipe down to close</Text>}
+    </View>
+  );
+
+  const renderHeader = () => <View style={styles.header} />;
+
+  return (
+    <>
```
Comment (Mykhailo-Kaduk) [inline src/components/BottomSheetModal/BottomSheetModal.tsx:71]: you don't need fragment here

File: src/components/BottomSheetModal/BottomSheetModal.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    });
+  }
+
+  const renderContent = () => (
+    <View style={styles.content}>
+      {children || <Text>Swipe down to close</Text>}
+    </View>
+  );
+
+  const renderHeader = () => <View style={styles.header} />;
```
Comment (Mykhailo-Kaduk) [inline src/components/BottomSheetModal/BottomSheetModal.tsx:68]: useCallback here

File: src/components/BottomSheetModal/BottomSheetModal.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  isOpen,
+  snapPoints,
+  currentPoint,
+  onClose,
+}) => {
+  const sheetRef = React.useRef<BottomSheet>(null);
+  const [isOpenBottomSheet, setBottomSheetState] = useState(false);
+
+  const [modalBackground] = useAnimatedInterpolation(isOpenBottomSheet, [
+    'rgba(131,131,131,0)',
```
Comment (Mykhailo-Kaduk) [inline src/components/BottomSheetModal/BottomSheetModal.tsx:36]: consider moving it to theme

File: src/components/BottomSheetModal/BottomSheetModal.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+  useEffect(() => {
+    if (isOpen) {
+      requestAnimationFrame(() => {
+        sheetRef.current?.snapTo(1);
+      });
+    }
+  }, [isOpen]);
+
+  function closeModal() {
```
Comment (stoyar) [inline src/components/BottomSheetModal/BottomSheetModal.tsx:53]: useCallback?

File: src/components/BottomSheetModal/BottomSheetModal.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    sheetRef.current?.snapTo(0);
+  }
+
+  function toggleBottomSheet() {
```
Comment (stoyar) [inline src/components/BottomSheetModal/BottomSheetModal.tsx:57]: The same

File: src/components/BottomSheetModal/styles.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    borderWidth: 1,
+    height: Sizes.BORDER_RADIUS,
+    borderTopLeftRadius: Sizes.BORDER_RADIUS,
+    borderTopRightRadius: Sizes.BORDER_RADIUS,
+    borderColor: Colors.GRAY,
+    borderBottomWidth: 0,
+    backgroundColor: Colors.WHITE,
+  },
+  content: {
+    padding: 16,
```
Comment (stoyar) [inline src/components/BottomSheetModal/styles.ts:15]: Use constants, please.

File: package.json [hunk 7]
```diff
@@ -45,6 +45,7 @@
     "react-native-screens": "^2.11.0",
     "react-native-vector-icons": "^7.1.0",
     "react-redux": "^7.2.1",
+    "reanimated-bottom-sheet": "^1.0.0-alpha.22",
```
[the remark this answers is not in the document]
Comment (andriibodryi) [reply package.json:48]: ok, I used this lib on prev project

**Review by stoyar — APPROVED** [review]

File: src/components/BottomSheetModal/BottomSheetModal.tsx [hunk 8]
```diff
@@ -0,0 +1,92 @@
+import React, { memo, useCallback, useEffect, useState } from 'react';
+import { Modal, TouchableOpacity, View } from 'react-native';
+import BottomSheet from 'reanimated-bottom-sheet';
```
Comment (Mykhailo-Kaduk) [inline src/components/BottomSheetModal/BottomSheetModal.tsx:3]: Could you please import it from components?

File: src/components/BottomSheetModal/BottomSheetModal.tsx [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+import { Text } from '~/components/Text/Text';
```
Comment (Mykhailo-Kaduk) [inline src/components/BottomSheetModal/BottomSheetModal.tsx:4]: just import from '~/components'

File: src/store/requestStatuses/selectors.ts [hunk 10]
```diff
@@ -1,3 +1,4 @@
+import { useSelector } from 'react-redux';
```
Comment (Mykhailo-Kaduk) [inline src/store/requestStatuses/selectors.ts:1]: import from store

File: src/store/requestStatuses/selectors.ts [hunk 11]
```diff
@@ -1,3 +1,4 @@
...
 import lodashGet from 'lodash/get';
```
Comment (Mykhailo-Kaduk) [inline src/store/requestStatuses/selectors.ts:2]: please create utils folder

File: src/store/requestStatuses/selectors.ts [hunk 12]
```diff
@@ -1,3 +1,4 @@
...
 import { AnyAction } from 'redux';
```
Comment (Mykhailo-Kaduk) [inline src/store/requestStatuses/selectors.ts:3]: import from store

**Review by Mykhailo-Kaduk — APPROVED** [review]

**Pull request merged**